mrbean/src/main/java/com/fasterxml/jackson/module/mrbean/TypeDefinitionUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.fasterxml.jackson.module.mrbean;
+
+import com.fasterxml.jackson.databind.JavaType;
+import net.bytebuddy.description.type.TypeDefinition;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.description.type.TypeDescription.Generic;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+
+final class TypeDefinitionUtil {
```

Comment:
This is a good start, although I suspect it might run into trouble for nested generic types and/or due to Jackson's bit inconsistent definition of generics for `Map`s and `Collection`s (for which it exposes generics used for binding `Map` / `Collection` and not to specific subtype).
But we can cross that bridge when get there since I doubt existing code handles these any better.

Reply:
Although I didn't run into any specific issues, I am sure that the are some obscure cases where the simple implementation I provided would stumple